tests.rb

```diff
@@ -0,0 +1,167 @@
+def check_validation(title, expected_error_message, &block)
+  print title
+  block.call
+  puts " -> FAIL!!"
+rescue ValidationError => error
+  result = error.message == expected_error_message
+  puts result ? ' -> Pass' : ' -> FAIL!!!'
+end
+
+def run_validation_tests
+  library = Library.new
+  author = Author.new('10', '1')
+  book = Book.new('1', '1')
+  reader = Reader.new('name1', 'email1', 'city1', 'street1', 1)
+  order = Order.new('0', '0')
+  library.add_book(book)
+  library.add_author(author)
+  library.add_order(order)
+  library.add_reader(reader)
+  puts '**************************************'
+  check_validation("Test validation for book title is nil", "title must be given") do
+    Book.new(nil, 'author')
+  end
+
+  check_validation("Test validation for book title is String", "title must be string") do
+    Book.new(Array.new, 'author')
+  end
+
+  check_validation("Test validation for book title is empty", "title must be not empty") do
+    Book.new('', 'author')
+  end
+  puts '*****************************************'
+
+  puts '**************************************'
+  check_validation("Test validation for book author is nil", "author must be given") do
+    Book.new('title', nil)
+  end
+
+  check_validation("Test validation for book author instance of Author", "author must be an instance of Author") do
+    Book.new('title', author)
+  end
+  puts '*****************************************'
+
+  puts '**************************************'
+  check_validation("Test validation for author name is nil", "name must be given") do
+    Author.new(nil, 'biography')
+  end
+
+  check_validation("Test validation for author name is String", "name must be string") do
+    Author.new(Array.new, 'biography')
+  end
+
+  check_validation("Test validation for author name is empty", "name must be not empty") do
+    Author.new('', 'biography')
+  end
+  puts '*****************************************'
+
+  puts '**************************************'
+  check_validation("Test validation for author biography is String", "biography must be string") do
+    Author.new('name', Array.new)
+  end
+
+  check_validation("Test validation for author biography is empty", "biography must be not empty") do
+    Author.new('name', '')
+  end
+  puts '*****************************************'
+
+  puts '**************************************'
+  check_validation("Test validation for order book is nil", "book must be given") do
```

Comment:
Fix order validation tests, they a wrong. You should pass correct params for each test case.